server/src/main/scala/services/OptimiserPlus.scala

```diff
@@ -0,0 +1,387 @@
+package services
+
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+case class OptimiserConfig(sla: Int, processors: WorkloadProcessorsLike)
+
+object OptimiserPlus {
```

Comment:
do you think we could come up with a better name than this? I don't have one, but just think `OptimiserPlus` doesn't tell you how it's different to `Optimiser`

Reply:
`OptimiserWithFlexibleProcessors`?